extension/src/popup/components/mnemonicPhrase/ConfirmMnemonicPhrase/index.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from "react";
+import { Redirect } from "react-router-dom";
+import { Formik } from "formik";
+import { useSelector, useDispatch } from "react-redux";
+
+import { APPLICATION_STATE } from "@shared/constants/applicationState";
+import {
+  confirmMnemonicPhrase,
+  authErrorSelector,
+  applicationStateSelector,
+} from "popup/ducks/accountServices";
+import { ROUTES } from "popup/constants/routes";
+import { Form, ApiErrorMessage, FormRow } from "popup/basics/Forms";
+import { FullscreenStyle } from "popup/components/FullscreenStyle";
+
+import { Button, Card } from "@stellar/design-system";
+
+import { CheckButton } from "../CheckButton";
+
+import "./styles.scss";
+
+const convertToWord = (wordKey: string) => wordKey.replace(/-.*/, "");
+
+export const ConfirmMnemonicPhrase = ({
+  words = [""],
+}: {
+  words: string[];
+}) => {
+  const dispatch = useDispatch();
+
+  const initialWordState = words.reduce(
+    (obj, current, i) => ({
+      ...obj,
+      // tag each word with an index because words can repeat
+      [`${current}-${i}`]: false,
+    }),
+    {},
+  );
+  const [selectedWords, setSelectedWords] = useState<string[]>([]);
+  const authError = useSelector(authErrorSelector);
+  const applicationState = useSelector(applicationStateSelector);
+
+  const updatePhrase = (target: HTMLInputElement) => {
+    if (target.checked) {
+      return setSelectedWords((prevState) => [...prevState, target.name]);
+    }
+    return setSelectedWords((prevState) => {
+      const currentArr = [...prevState];
+      currentArr.splice(currentArr.indexOf(target.name), 1);
+      return [...currentArr];
+    });
+  };
+
+  const wordStateArr: [string, boolean][] = Object.entries(initialWordState);
+
+  const handleSubmit = async () => {
+    await dispatch(confirmMnemonicPhrase(displaySelectedWords()));
+  };
+
+  const displaySelectedWords = () =>
+    selectedWords.map((word) => convertToWord(word)).join(" ");
+
+  if (applicationState === APPLICATION_STATE.MNEMONIC_PHRASE_CONFIRMED) {
+    return <Redirect to={ROUTES.mnemonicPhraseConfirmed} />;
+  }
+
+  return (
+    <>
+      <FullscreenStyle />
+      <div className="ConfirmMnemonicPhrase--screen">
```

Comment:
NIT, in [BEM naming convention ](http://getbem.com/naming/), `--` is for a modifier. I think in most of these cases we are using them for block elements `__`. It's not like we're 100% consistent everywhere. :sweat_smile:

Reply:
til! https://github.com/stellar/freighter/pull/359/commits/4a93c073f0e2d410ff35585a1da3f851e41484c4